graphbolt/src/cuda/extension/unique_and_compact_map.cu

```diff
@@ -25,11 +25,10 @@
 
 #include <cub/cub.cuh>
 
-#ifdef GRAPHBOLT_USE_ROCM
+#include <cuco/static_map.cuh>
 // libhipcxx does not provide proclaim_return_type, but hipCollections defines
 // it.
```

Comment:
This comment goes with the include above it.

Also, an extra include is harmless, but based on the previous ifdef guard, I think it was not required by CUDA. I'm okay with simplifying the logic or staying closer to include-what-you-use with the guard.

Reply:
If it works then it looks good. What I think I remember is that previously:
`cuda/functional` with cuda provides `proclaim_return_type`. With HIP/ROCm, `/functional` does not have that functionality, but `hipCollections` (`cuco`) does. It was for compile errors though, so if it compiles on both, it should be fine.

The [upstream version](https://github.com/dmlc/dgl/blob/743e65fa3ba9781bef443bfdbca7d87983e1071f/graphbolt/src/cuda/cooperative_minibatching_utils.cu#L25) doesn't have that include, so it does seem like something is wrong. The only thing I could think of though is if GRAPHBOLT_USE_ROCM is defined for Nvidia also which would break other pieces of our changes.

Reply:
Ok, no `cuda/functional` is missing with hip. So I think what I've got now is correct.

Reply:
I updated the comment though. I can't really elaborate on what symbols are needed from this and where the hip version is getting them instead because the cuda version actually builds fine without this include. Maybe it's getting it transitively from somewhere else.